### ISSUE TITLE
Any plans for an @dispatch decorator?

### ISSUE THREAD

**Issue by jspengeman** [angular-redux/store#384]
I saw in [this](http://blog.rangle.io/getting-started-with-redux-and-angular-2/) blog post by @e-schultz that the ```@select``` and ```@dispatch``` features were on the roadmap (authored a year ago). I see that select is implemented but was curious if there are plans to implement the dispatch decorator that the blog post mentions?

**Comment by e-schultz** [comment]
Dispatch was on the roadmap, but fell off of my radar for awhile.  I do think it's worth revisiting though.

Initially, I was moving towards favouring `ActionServices` that did the dispatch for you instead of just returning the POJO, but starting to rethink that.

Given:

```ts
export class SomeComponent {
  constructor(private actions:Actions) { }
   @dispatch()
   something() {
       return actions.myActionCreator('a','b','c');
   }
}
```

I would see dispatch being called with the result - but the result still getting returned. This would fit the use-case of your action-creator being a service for whatever reason. 

Could also possibly do something like:

```ts
import { myActionCreator } from './somePlace';
@Component({
template: `<button (click)="myAction('hello!')">Click</button>`
});
export class MyComponent {

   @dispatch()
   myAction = myActionCreator;
}
```
This could work in the case here your action creators are just pure/exported functions that don't need access to Angular DI.

If you had a case where you didn't need to do any logic within the component, and the template called `myAction(x,y,z)` directly.

Not opposed to adding this, any thoughts / use cases I might be missing? @SethDavenport

**Comment by e-schultz** [comment]
@jspengeman have an initial pass of this as a pr - going to do some more testing, but shouldn't be too long until it's ready.

**Comment by jspengeman** [comment]
@e-schultz Thanks for the detailed reply. Additionally, thanks getting something going so quickly. I would have contributed but I was curious what the maintainers thoughts were first. Appreciate the initiative. 

Ill take a look at the PR.

**Comment by SethDavenport** [comment]
I like this - injecting NgRedux and dispatching manually in ActionServices is convenient but feels a bit weird to write.

Doing an `@dispatch` on a property like this is a lot nicer:

```typescript
@Injectable()
export class ActionService {
  @dispatch myAction = (payload) => ({ type: 'MINE', payload })
  @dispatch myActionToo = (payload) => ({ type: 'MINE2', payload })
  @dispatch myThirdAction = (payload) => ({ type: 'MINE3', payload })
}
``

(as long as AoT can handle those arrow functions, I haven't tried)

Whatever we do, it should allow people to easily unit test their action creators though.

**Comment by SethDavenport** [comment]
`@dispatch` has been released as part or 6.3.0.
